File: EngineKitGame/EngineKit/EngineKitTests/TestCommons.m


#import "TestCommons.h"


void setupRandomSeed() {
    NSDateComponents *components = [[NSCalendar currentCalendar]
                                    components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear
                                    fromDate:[NSDate date]];

    unsigned int seed = (unsigned int)components.day
    + (unsigned int)components.month * 31
    + (unsigned int)components.year * 12 * 31;

    srand(seed);
}


CGFloat randomFloat() {

    CGFloat f = ((double)rand() / rand()) - ((double)rand() / rand());
    f = f * f * f;

    if (f == 0) return randomFloat();

    return f;
}

File: EngineKit/EngineKit/TestCommons.swift


import Foundation
import XCTest


func setupRandomSeed() {
    let calendar = NSCalendar.currentCalendar()
    let units: NSCalendarUnit = .DayCalendarUnit |
        .MonthCalendarUnit |
        .YearCalendarUnit
    let components = calendar.components(units, fromDate: NSDate())

    let seed = components.day + components.month * 31 + components.year * 12*31

    srand(UInt32(seed))
}


func floatRand() -> Float {
    return Float(rand())
}


func randomFloat() -> Float {
    var f = (floatRand() / floatRand()) - (floatRand() / floatRand())
    f = f * f * f

    if (f == 0) {
        return randomFloat()
    }

    return f
}

